### ISSUE TITLE
In certain cases, cant use '_'  at new_post_name.

### ISSUE THREAD

**Issue by tetsuyainfra** [hexojs/hexo#3862]
## Check List
- [x] I have already read [Docs page](https://hexo.io/docs/) & [Troubleshooting page](https://hexo.io/docs/troubleshooting)
- [x] I have already searched existing issues and they are not help  to me
- [x] I examined error or warning messages and it's difficult to solve
- [x] Using [the latest](https://github.com/hexojs/hexo/releases) version of Hexo (run `hexo version` to check)
- [x] Node.js is higher than 8.6.0

## Expected behavior
`INFO  Created: ~/work/jisaku-club/blog/source/_posts/2019-11-19_test.md
## Actual behavior`
`INFO  Created: ~/work/jisaku-club/blog/source/_posts/2019-11-undefinedtest.md`
## How to reproduce?

* Change _config.yaml as bellow
* hexo new post test

## Environment & Settings

**Node.js & npm version**

```
node v12.13.0
npm 6.13.0
```

**Your site `_config.yml`** (Optional)

```
new_post_name: ':year-:month-:day_:title.md' # File name of new posts
```

**Hexo and Plugin version(`npm ls --depth 0`)**

```
hexo-site@0.0.0 /home/tetsuyainfra/work/jisaku-club/blog
├── hexo@4.0.0
├── hexo-deployer-git@2.0.0
├── hexo-generator-archive@1.0.0
├── hexo-generator-category@1.0.0
├── hexo-generator-index@1.0.0
├── hexo-generator-tag@1.0.0
├── hexo-renderer-ejs@1.0.0
├── hexo-renderer-marked@2.0.0
├── hexo-renderer-stylus@1.1.0
└── hexo-server@1.0.0
```

**Your package.json `package.json`**

```
{                                                 
  "name": "hexo-site",                            
  "version": "0.0.0",                             
  "private": true,                                
  "scripts": {                                    
    "build": "hexo generate",                     
    "clean": "hexo clean",                        
    "deploy": "hexo deploy",                      
    "server": "hexo server"                       
  },                                              
  "hexo": {                                       
    "version": "4.0.0"                            
  },                                              
  "dependencies": {                               
    "hexo": "^4.0.0",                             
    "hexo-deployer-git": "^2.0.0",                
    "hexo-generator-archive": "^1.0.0",           
    "hexo-generator-category": "^1.0.0",          
    "hexo-generator-index": "^1.0.0",             
    "hexo-generator-tag": "^1.0.0",               
    "hexo-renderer-ejs": "^1.0.0",                
    "hexo-renderer-marked": "^2.0.0",             
    "hexo-renderer-stylus": "^1.1.0",             
    "hexo-server": "^1.0.0"                       
  }                                               
}                                                 
```

## Others
When I followed the code, I found out that I could change regex key will be ok.
https://github.com/hexojs/hexo-util/blob/master/lib/permalink.js#L5

But present code, variable keys already include '_'.
https://github.com/hexojs/hexo/blob/eb33af08960ba29ba458273851e084ffd0455d68/lib/plugins/filter/new_post_path.js#L10-L17

I thought I could use the current key to regular expression,
but it matter if the number of keys increases in the future.

I couldn't come up with a way to solve it.

**Comment by seaoak** [comment]
@tetsuyainfra 's comment is right.
I think this behavior is a bug of `hexo-util/lib/permalink.js`.

I make a new patch hexojs/hexo-util#139.

**Comment by curbengh** [comment]
Fix has been merged, please wait its release in hexo-util [1.6.0](https://github.com/hexojs/hexo-util/pull/133).
